evalml/automl/automl_search.py

```diff
@@ -318,6 +317,19 @@ def __init__(self,
                                        num_ensemble_batches)
             else:
                 self.max_iterations = 1 + len(self.allowed_pipelines) + (self._pipelines_per_batch * (self.max_batches - 1))
+        if run_ensembling:
+            if not (0 < _ensembling_split_size < 1):
+                raise ValueError(f"Ensembling split size must be between 0 and 1 exclusive, received {_ensembling_split_size}")
+            _, self.ensembling_indices = next(TrainingValidationSplit(test_size=_ensembling_split_size, shuffle=True, stratify=self.y_train.to_series(), random_state=self.random_seed).split(self.X_train.to_dataframe(), self.y_train.to_series()))
```

Comment:
@bchen1116 yep, computing the indices here and passing them to the engine looks good to me!

Could you use our `split_data` helper here instead? That will do a standard TV split for regression data, a stratified split for classification data and an ordered TV split for timeseries data. We should use that any time we want to do a two-part split. Otherwise we could be giving the ensembler a biased or unfair split!

Also just as a stylistic aside, its definitely slick to do this in one line, but IMO its preferable to split it into multiple lines, so others can quickly and easily understand what's up. We don't have a max line length enforced in our linter right now, but, yeah! :)